app/Http/Controllers/Api/V1/GoogleClassroomController.php

```diff
@@ -136,108 +139,119 @@ public function saveAccessToken(Request $request){
+                        continue;
+                    }
                     $activity->shared = true;
                     $activity->save();
 
                     //$updated_activity = Activity::where(['_id'=>$activity->_id])->update(['shared'=> true]);
 
-
-                    $courseWork = new \Google_Service_Classroom_CourseWork();
-                    $courseWork->setCourseId($course->id);
-                    $courseWork->setTopicId($topic->topicId);
-                    /*$h5p_activity = \DB::connection('mysql')
-                            ->table('h5p_contents')
-                            ->select('title')
-                            ->where(['id'=>$activity->mysqlid])->first();
-                    $courseWork->setTitle($h5p_activity->title);*/
-                    $courseWork->setTitle($activity->title);
-                    $courseWork->setWorkType('ASSIGNMENT');
-                    $courseWork->setMaterials([
-                        'link'=> [
-                            'url' => config('constants.front-url').'/shared/activity/'.$activity->id
-                        ]
-                    ]);
-                    $courseWork->setState('PUBLISHED');
-
-                    $courseWork = $service->courses_courseWork->create($course->id, $courseWork);
-                    $return['course']['topics'][$count]['courseWork'] = [];
-                    $return['course']['topics'][$count]['courseWork'][$courseWorkCount] = (array) $courseWork;
+                    $courseWorkData = [
+                        'course_id' => $course->id,
+                        'topic_id' => $topic->topicId,
+                        'activity_id' => $activity->id,
+                        'activity_title' => $activity->title
+                    ];        
+                    $courseWork = $service->createCourseWork($courseWorkData);
+                    
+                    $return['course']['topics'][$count]['course_work'] = [];
+                    $return['course']['topics'][$count]['course_work'][$courseWorkCount] = GCCourseWorkResource::make($courseWork)->resolve();
                     $courseWorkCount++;
                 }
                 $count++;
             }
             
+            return response([
+                'data'=> $return,
+                'status' => 'success'
+            ], 200);
 
-            return responseSuccess($return);
-            
         } catch(\Google_Service_Exception $ex){
-            return responseFail(['message' => $ex->getMessage()]);
+            return response([
+                'errors' => [$ex->getMessage()],
+            ], 500);
         } catch (\Exception $ex){
-            return responseFail(['message' => $ex->getMessage()]);
+            return response([
+                'errors' => [$ex->getMessage()],
+            ], 500);
+        }
+    }
+
+    private function checkPermission(Project $project)
+    {
+        $authenticated_user = auth()->user();
```

Comment:
variable case

Reply:
I took out this function, as I used a Gate instead.

Reply:
are we ok this is